tests/resources/event_gateway_reusable_test.go

```diff
@@ -0,0 +1,298 @@
+		builder := hclbuilder.NewWithProvider(hclbuilder.KonnectBeta, fmt.Sprintf(providerConfigTemplate, serverScheme, serverHost, serverPort))
+		builder.ProviderProperty = hclbuilder.KonnectBeta
+
+		egwCp, err := hclbuilder.FromString(`
+			resource "konnect_event_gateway" "tf_test_event_gateway" {
+  				name = "my_test_event_gateway"
+			}
+		`)
+		require.NoError(t, err)
+
+		egwBackendCluster, err := hclbuilder.FromString(`
+		resource "konnect_event_gateway_backend_cluster" "tf_test_egw_backend_cluster" {
+			name = "tf_test_egw_backend_cluster"
+			authentication = {
+    			anonymous = {}
+  			}
+			bootstrap_servers = [
+				"10.0.0.1:8080"
+			]
+			tls = {
+				enabled = false
+				insecure_skip_verify = true
+			}
+		}
+		`)
+		require.NoError(t, err)
+
+		// Set up dependencies
+		egwBackendCluster.AddAttribute("gateway_id", egwCp.ResourcePath()+".id")
+
+		resource.Test(t, resource.TestCase{
+			ProtoV6ProviderFactories: providerFactory,
+			Steps: []resource.TestStep{
+				{
+					Config: builder.Upsert(egwCp).Upsert(egwBackendCluster).Build(),
+					ConfigPlanChecks: resource.ConfigPlanChecks{
+						PreApply: []plancheck.PlanCheck{
+							plancheck.ExpectResourceAction("konnect_event_gateway.tf_test_event_gateway", plancheck.ResourceActionCreate),
+							plancheck.ExpectResourceAction("konnect_event_gateway_backend_cluster.tf_test_egw_backend_cluster", plancheck.ResourceActionCreate),
+						},
+					},
+					Check: resource.ComposeAggregateTestCheckFunc(
+						resource.TestCheckResourceAttr("konnect_event_gateway_backend_cluster.tf_test_egw_backend_cluster", "name", "tf_test_egw_backend_cluster"),
+					),
+				},
+				{
+					Config: builder.Upsert(egwCp).Upsert(egwBackendCluster).Build(),
+					ConfigPlanChecks: resource.ConfigPlanChecks{
+						PreApply: []plancheck.PlanCheck{
+							plancheck.ExpectEmptyPlan(),
+						},
+					},
+				},
+				{
+					Config: builder.Upsert(egwCp).Upsert(egwBackendCluster.AddAttribute("labels", `{key = "value"}`)).Build(),
+					Check: resource.ComposeAggregateTestCheckFunc(
+						resource.TestCheckResourceAttr("konnect_event_gateway_backend_cluster.tf_test_egw_backend_cluster", "labels.%", "1"),
+						resource.TestCheckResourceAttr("konnect_event_gateway_backend_cluster.tf_test_egw_backend_cluster", "labels.key", "value"),
+					),
+				},
+				{
+					Config: builder.Upsert(egwCp).Build(),
+					ConfigPlanChecks: resource.ConfigPlanChecks{
+						PreApply: []plancheck.PlanCheck{
+							plancheck.ExpectEmptyPlan(),
+						},
+					},
+				},
+			},
+		})
+	})
+
+	t.Run("EGW Virtual Cluster", func(t *testing.T) {
+		builder := hclbuilder.NewWithProvider(
+			hclbuilder.KonnectBeta,
+			fmt.Sprintf(providerConfigTemplate, serverScheme, serverHost, serverPort),
+		)
+		builder.ProviderProperty = hclbuilder.KonnectBeta
+
+		// Event Gateway (Control Plane)
+		egwCp, err := hclbuilder.FromString(`
+resource "konnect_event_gateway" "tf_test_event_gateway" {
```

Comment:
@shivaygupta-dotcom couple of things - 
1. Lets make use of indentation for these as well, makes it easier to read.
2. Since we are repeating the parent resources -  like `konnect_event_gateway` - can we create them once and reuse? We can template them using `fmt.Sprintf` if needed.
3. Let's limit adding comments only when code is not obviuous